server/src/test/java/org/candlepin/controller/PoolManagerFunctionalTest.java

```diff
@@ -746,5 +731,127 @@ public void testBatchBindZeroQuantity() throws EntitlementRefusedException {
         assertEquals(1, results.get(1).getQuantity().intValue());
     }
 
+    @Test
+    public void testCleanupExpiredPools() {
+        long ct = System.currentTimeMillis();
+        Date activeStart = new Date(ct + 3600000);
+        Date activeEnd = new Date(ct + 7200000);
+        Date expiredStart = new Date(ct - 7200000);
+        Date expiredEnd = new Date(ct - 3600000);
+
+        Owner owner = this.createOwner();
+        Product product1 = this.createProduct(owner, "test-product-1", "Test Product 1");
+        Product product2 = this.createProduct(owner, "test-product-2", "Test Product 2");
+        Pool activePool = this.createPool(owner, product1, 1L, activeStart, activeEnd);
+        Pool expiredPool = this.createPool(owner, product2, 1L, expiredStart, expiredEnd);
+
+        try {
+            this.beginTransaction();
```

Comment:
@nguyenfilip I suspect this was because @Transaction was being ignored, cause the method was private. this should have been a red flag to me. any who, fixed now.